internal/graph/graph.go

```diff
@@ -213,3 +214,47 @@ func (g *Graph[V]) TopologicallySortWithPriority(isLowerPriority func(V, V) bool
 
 	return output, nil
 }
+
+func (g *Graph[V]) EncodeDOT(w io.Writer, sortVertices bool) (err error) {
```

Comment:
Rather than this being a method of a `Graph`, can it instead, be a function:
`func EncodeDot(graph *Graph[V], sortVertices bool)`. It will help limit the scope of the Graph a bit. 

I'd put the function in the same file as the `DotBuilder`, since they go hand-in-hand, and maybe call the file `encode_dot.go`

Small nit: Get rid of the named parameter. This can lead to weird bugs/behavior within go and idelly should be avoided when possible